presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/CteUtils.java

```diff
@@ -1,76 +0,0 @@
-/*
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package com.facebook.presto.sql.planner.optimizations;
-
-import com.facebook.presto.common.type.RowType;
-import com.facebook.presto.common.type.VarcharType;
-import com.facebook.presto.common.type.Varchars;
-import com.facebook.presto.spi.PrestoException;
-import com.facebook.presto.spi.relation.VariableReferenceExpression;
-
-import java.util.List;
-
-import static com.facebook.presto.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
-
-public class CteUtils
```

Comment:
Very nice!

Reply:
@tdcmeehan  I just saw and one test case is failing
 ```
@Test
    public void testCteWithZeroLengthVarchar()
    {
        String testQuery = "WITH temp AS (" +
                "  SELECT * FROM (VALUES " +
                "    (CAST('' AS VARCHAR(0)), 9)" +
                "  ) AS t (text_column, number_column)" +
                ") SELECT * FROM temp";
        QueryRunner queryRunner = getQueryRunner();
        compareResults(queryRunner.execute(getMaterializedSession(),
                        testQuery),
                queryRunner.execute(getSession(),
                        testQuery));
    }
```

Stack Trace
```
Caused by: java.lang.RuntimeException: Varchar length 0 out of allowed range [1, 65535]
	at org.apache.hadoop.hive.serde2.typeinfo.BaseCharUtils.validateVarcharParameter(BaseCharUtils.java:32)
	at org.apache.hadoop.hive.serde2.typeinfo.VarcharTypeInfo.<init>(VarcharTypeInfo.java:33)
	at org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory.createPrimitiveTypeInfo(TypeInfoFactory.java:159)
	at org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory.getPrimitiveTypeInfo(TypeInfoFactory.java:117)
	at org.apache.hadoop.hive.serde2.typeinfo.TypeInfoFactory.getVarcharTypeInfo(TypeInfoFactory.java:183)
	at com.facebook.presto.hive.HiveTypeTranslator.translate(HiveTypeTranslator.java:98)
	at com.facebook.presto.hive.HiveType.toHiveType(HiveType.java:218)
	at com.facebook.presto.hive.HiveMetadata.getColumnHandles(HiveMetadata.java:3584)
	at com.facebook.presto.hive.HiveMetadata.createTemporaryTable(HiveMetadata.java:1107)


```


So we might need this check for the 0 length Varchar due to , because  write with 0 length varchar fails.

See reference - While Presto supports Varchar of length 0 (as discussed in https://github.com/trinodb/trino/issues/1136

Reply:
I am hence planning to disallow cte creation if there is a 0 length varchar

Reply:
But that issue should exist in ctas as well i think, will differ this test case into a new issue

Reply:
Created #21791 and adding todos in test

Reply:
Confirmed that ```
CREATE TABLE tmp.test2512351 AS (
    SELECT * FROM (VALUES 
        (CAST('' AS VARCHAR(0)), 9)
    ) AS t(col1, col2)
);
```
fails in Presto